photon-api/src/main/scala/com/linkedin/photon/ml/estimators/GameEstimator.scala

```diff
@@ -49,119 +53,160 @@ import com.linkedin.photon.ml.util._
  * @param sc The spark context for the application
  * @param logger The logger instance for the application
  */
-class GameEstimator(val sc: SparkContext, implicit val logger: Logger) {
+class GameEstimator(val sc: SparkContext, implicit val logger: Logger) extends Params {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

It's difficult to see on GitHub, so you may want to pull the patch locally. Major changes:
- `var` parameters are replace with `spark.ml.param.Param` parameters which are stored within a `ParamMap`
- parameters have validation functions, but there is also a `validateParams` function to check interactions between parameters
- The old explicit `GameModelOptimizationConfiguration` has been replaced by a generic map of `CoordinateID -> CoordinateOptimizationConfiguration`